sql/core/src/main/scala/org/apache/spark/sql/execution/BaseScriptTransformationExec.scala

```diff
@@ -220,6 +226,9 @@ trait BaseScriptTransformationExec extends UnaryExecNode {
       case CalendarIntervalType => wrapperConvertException(
         data => IntervalUtils.stringToInterval(UTF8String.fromString(data)),
         converter)
+      case _: ArrayType | _: MapType | _: StructType =>
+        wrapperConvertException(data => JsonToStructs(attr.dataType, Map.empty[String, String],
```

Comment:
This can cause much overhead cuz this make a new object (`JsonToStructs `) for each call. Could you avoid it?

Reply:
> This can cause much overhead cuz this make a new object (`JsonToStructs `) for each call. Could you avoid it?

This problem also happen in input side's `Cast` and `StructToJson`.
To avoid it maybe we need to extract common method from these expression or just write some thing for` ScriptTransform`.  WDYT @cloud-fan

Reply:
> This problem also happen in input side's Cast and StructToJson.

Really? I think the input side does not have the issue. Why do you think so?

Reply:
> > This problem also happen in input side's Cast and StructToJson.
> 
> Really? I think the input side does not have the issue. Why do you think so?

Hmmm, I got your point. I misunderstood you point. Updated, how about current?  Test with df.repartition(1) then transform. It only create JsonToStruct once for one partition.